ml-algorithms/src/main/java/org/opensearch/ml/engine/MLEngine.java

```diff
@@ -132,13 +139,21 @@ public MLOutput trainAndPredict(Input input) {
         return trainAndPredictable.trainAndPredict(mlInput);
     }
 
-    public Output execute(Input input) {
+    public Output execute(Input input) throws Exception {
         validateInput(input);
-        Executable executable = MLEngineClassLoader.initInstance(input.getFunctionName(), input, Input.class);
-        if (executable == null) {
-            throw new IllegalArgumentException("Unsupported executable function: " + input.getFunctionName());
+        if (input.getFunctionName() == FunctionName.METRICS_CORRELATION) {
```

Comment:
Throw `IllegalArgumentException` for other cases?

Reply:
By `other cases`, you mean for other executable? Yeah they are in the other `else` condition with previous code block:

```
            Executable executable = MLEngineClassLoader.initInstance(input.getFunctionName(), input, Input.class);
            if (executable == null) {
                throw new IllegalArgumentException("Unsupported executable function: " + input.getFunctionName());
            }
            return executable.execute(input);
```

Reply:
Got it, thanks, then no need to throw exception, missed the else block when first review